src/util/requests/impl/WsConnectionImpl.hpp

```diff
@@ -92,6 +117,31 @@ class WsConnectionImpl : public WsConnection {
             return RequestError{"Close error", errorCode};
         return std::nullopt;
     }
+
+private:
+    template <typename Operation>
+    static void
+    withTimeout(Operation&& operation, boost::asio::yield_context yield, std::chrono::steady_clock::duration timeout)
+    {
+        boost::asio::cancellation_signal cancellationSignal;
+        auto cyield = boost::asio::bind_cancellation_slot(cancellationSignal.slot(), yield);
+
+        boost::asio::steady_timer timer{boost::asio::get_associated_executor(cyield), timeout};
+        timer.async_wait([&cancellationSignal](boost::system::error_code errorCode) {
+            if (!errorCode)
+                cancellationSignal.emit(boost::asio::cancellation_type::terminal);
+        });
+        operation(cyield);
+    }
+
+    static boost::system::error_code
+    mapError(boost::system::error_code const ec)
```

Comment:
Do we add 'const' for value parameter?

Reply:
I think it is useful, showing that `ec` will not be changed.